ocs_ci/framework/__init__.py

```diff
@@ -451,6 +452,21 @@ def switch_to_cluster_by_cluster_type(self, cluster_type, num_of_cluster=0):
             self.get_cluster_type_indices_list(cluster_type)[num_of_cluster]
         )
 
+    def run_for_all_clusters(self, func):
+        """
+        A decorator to run the decorated function for all clusters
+        and switch context between them.
+        """
+
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            for cluster_index in range(self.nclusters):
+                self.switch_ctx(cluster_index)
```

Comment:
We will need to use context manager which is switching back to original context or store original context and once function will finish, switch back to original context.